### ISSUE TITLE
`VarCouldBeVal` false positive

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4474]
## Expected Behavior
If I run plain detekt (no type solving) with this code:

```kotlin
interface I {
    var optionEnabled: Boolean
}
fun test(): I {
    val o = object: I {
        override var optionEnabled: Boolean = false
    }
    return o
}
```

It shouldn't raise a `VarCouldBeVal` issue.

## Observed Behavior
It does

## Context
We shoudl accept the false-negatives in this issue without type-solving: any override `var` should be allowed if we don't have a proper `bindingContext`.

I created a test to demostrate it: 

```kotlin
it("does not report when an object initializes a variable directly - without type solving") {
    val code = """
        interface I {
            var optionEnabled: Boolean
        }
        fun test(): I {
            val o = object: I {
                override var optionEnabled: Boolean = false
            }
            return o
        }
    """
    assertThat(subject.compileAndLint(code)).isEmpty()
}
```
## Your Environment
* Version of detekt used: 1.17.1, 1.19.0 & current main

**Comment by severn-everett** [comment]
How would this rule be run if type resolution isn't enabled? I was under the impression that the `@RequiresTypeResolution` annotation meant that it shouldn't run in this scenario.

**Comment by BraisGabin** [comment]
I didn't check it, but it does. And it found some real cases (I runned it over https://github.com/mozilla-mobile/fenix) I assume that without type solving it just check private or local `var`s

**Comment by cortinico** [comment]
Fixed by 4477
